torch/csrc/api/src/cuda.cpp

```diff
@@ -52,12 +53,13 @@ void manual_seed_all(uint64_t seed) {
 
 void synchronize(int64_t device_index) {
   TORCH_CHECK(is_available(), "No CUDA GPUs are available");
-  int64_t num_gpus = cuda::device_count();
+  auto num_gpus = cuda::device_count();
   TORCH_CHECK(
-      device_index == -1 || device_index < num_gpus,
+      device_index < 0 || static_cast<size_t>(device_index) < num_gpus,
```

Comment:
Is device index being `-2` valid, seems like `-1` might be a special value here?

Reply:
No, we only accept >=0 indices.